src/App.tsx

```diff
@@ -14,11 +23,124 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isVisible: boolean,
+  visibleGoods: string[],
+};
 
+class App extends React.Component<{}, State> {
+  state = {
+    isVisible: false,
+    visibleGoods: goodsFromServer,
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>Goods</h1>
+        <Box sx={{ m: 2 }}>
+          Goods available: {goodsFromServer.length}
+        </Box>
+        <br />
+        {this.state.isVisible === false && (
+          <Button
+            type="button"
+            onClick={() => {
+              this.setState(
+                (state) => {
+                  const prev = state.isVisible;
+
+                  return ({
+                    isVisible: !prev,
+                  });
+                },
+              );
+            }}
+          >
+            Start
+          </Button>
+        )}
+
+        {this.state.isVisible && (
```

Comment:
Use destructuring: get needed values from this state before the return statement and use them in your code